src/interaction/interactive_api.jl

```diff
@@ -177,15 +177,22 @@ function screen_relative(scene::Scene, mpos)
 end
 
 """
-    mouseposition(scene = hovered_scene())
+    mouseposition(ax/scene)
```

Comment:
```suggestion
    mouseposition(scene::Scene)
    mouseposition(ax::Axis)
```
maybe?  Not necessary but perhaps clearer.

Scenes can also be arbitrarily transformed so we should probably implement these changes there as well!

Reply:
Fixed the docstring in 4abb55536, I'll work on adding the transforms for scenes. Do you have any pointers on how I could test that? I haven't had much luck with that so far.

Reply:
So if we were to apply this to all scenes then we wouldn't need to specialize on `Axis` and this would work:
```julia
mouseposition(x) = mouseposition(get_scene(x))

function mouseposition(scene::Scene = hovered_scene())
    pos = to_world(scene, mouseposition_px(scene))

    inv_transform = Makie.inverse_transform(scene.transformation.transform_func[])
    return Vec2{Float64}(Makie.apply_transform(inv_transform, pos))
end
```

That code gives the correctly transformed numbers, but I get some odd results:
![makie-scene-transform](https://github.com/user-attachments/assets/4af4c0c7-7b29-478e-aa5f-fac7e373db15)

I'm guessing there's code elsewhere that's assuming `mouseposition(ax)` won't apply the scene transform.

Reply:
Yep that's probably in the `select_[point/rectangle/line]` routines.  They might have worked in "camera" space (transformed space, so after `apply_transform` and `transformationmatrix`).

Reply:
I think those should actually be doing the right thing now because the line is being drawn correctly, but luckily the only other place in Makie.jl where `mouseposition()` is used is in the mouse state engine which seems to expect the untransformed position. Fixed that in 6622ec4cc but I don't know if the API is very good. Maybe it'd be better to have a separate function?